skyportal/handlers/api/spectrum.py

```diff
@@ -154,87 +140,106 @@ def post(self):
                 f'Invalid / missing parameters; {e.normalized_messages()}'
             )
 
-        # always append the single user group
-        single_user_group = self.associated_user_object.single_user_group
+        with self.Session() as session:
+            stmt = Instrument.select(self.current_user).where(
+                Instrument.id == data['instrument_id']
+            )
+            instrument = session.scalars(stmt).first()
```

Comment:
Maybe we can standardize on `session.execute(stmt).scalars().first()` or `session.execute(stmt).all()` etc. to make it a bit clearer what's going on.  (It's a bit weird that `session.scalars` executes a statement, so may be confusing to readers.)

Can happen in another PR.

Reply:
I think there may be a `scalars_first` and `scalars_all` that can do that as well. I'll check on that.